text/todo-generic-payload.md

```diff
@@ -0,0 +1,368 @@
+| 2     | GENERIC_PAYLOAD_SUPPORTS_MOMENTARY | Function supports momentary control mode. |
+
+#### GENERIC_PAYLOAD_VALUE_TYPE
+
+Data type for generic payload function values.
+
+| Value | Name                              | Description                     |
+| ----- | --------------------------------- | ------------------------------- |
+| 0     | GENERIC_PAYLOAD_VALUE_TYPE_INT32  | 32-bit signed integer.          |
+| 1     | GENERIC_PAYLOAD_VALUE_TYPE_UINT32 | 32-bit unsigned integer.        |
+| 2     | GENERIC_PAYLOAD_VALUE_TYPE_REAL32 | 32-bit IEEE-754 floating point. |
+| 3     | GENERIC_PAYLOAD_VALUE_TYPE_INT64  | 64-bit signed integer.          |
+| 4     | GENERIC_PAYLOAD_VALUE_TYPE_UINT64 | 64-bit unsigned integer.        |
+| 5     | GENERIC_PAYLOAD_VALUE_TYPE_REAL64 | 64-bit IEEE-754 floating point. |
+
+#### GENERIC_PAYLOAD_FUNCTION_TYPE
+
+Type of generic payload function.
+
+| Value | Name                                     | Description                                                                |
+| ----- | ---------------------------------------- | -------------------------------------------------------------------------- |
+| 0     | GENERIC_PAYLOAD_FUNCTION_TYPE_LOGICAL    | Logical state control (true/false, on/off, high/low).                      |
+| 1     | GENERIC_PAYLOAD_FUNCTION_TYPE_CONTINUOUS | Continuous value control (e.g., servo angle, motor speed). Uses min/max.   |
+| 2     | GENERIC_PAYLOAD_FUNCTION_TYPE_DISCRETE   | Discrete value selection (e.g., mode number, PWM frequency). Uses min/max. |
+
+#### GENERIC_PAYLOAD_ERROR_FLAGS
+
+Error flags for a generic payload. This is a bitmask enumeration.
+
+| Value | Name                                       | Description                                                                                                              |
+| ----- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------ |
+| 1     | GENERIC_PAYLOAD_ERROR_FLAGS_SOFTWARE_ERROR | There is an error with the generic payload's software.                                                                   |
+| 2     | GENERIC_PAYLOAD_ERROR_FLAGS_HARDWARE_ERROR | There is an error with the generic payload's hardware.                                                                   |
+| 4     | GENERIC_PAYLOAD_ERROR_FLAGS_OVERTEMP_FAULT | A specified limit has been reached by the generic payload's temperature sensor. Applies if payload monitors temperature. |
+| 8     | GENERIC_PAYLOAD_ERROR_FLAGS_CUSTOM         | Generic payload custom failure. See custom error flag bitmask for details.                                               |
+
+### Messages
+
+#### GENERIC_PAYLOAD_STATUS (Message ID: 60000)
+
+Status and discovery message for a generic payload. This should be broadcast at a low rate (nominally 1 Hz) to announce the payload's presence and capabilities.
```

Comment:
I would be tempted to suggest that this be split into static and dynamic info. Use the https://mavlink.io/en/services/gimbal_v2.html#discovery-of-gimbal-manager approach based on MAV_TYPE to discover the payloads.

Not entirely sure, because it mixes status and metadata. Also I'm still not sure on how functions and payloads map in the real world, so maybe this is sensible.

Reply:
i think that makes sense.